lib/logstash/plugin_mixins/elasticsearch/common.rb

```diff
@@ -192,6 +192,26 @@ def retrying_submit(actions)
       end
     end
 
+    # we shouldn't unsupported actions to ES
+    # method filters out unsupported actions by warning, writes event to DQL if enabled
+    # @returns valid actions
+    def filter_unsupported_actions(actions)
```

Comment:
_(unit tests will fail, confirming if this approach makes sense)_

@yaauie, as we took offline, time complexity becomes little extended but we overcome the order mismatch case. If you let me know your feedback I will test more and update the unit tests. Thanks.

Reply:
method naming nitpick:

The word `filter` is ambiguous; lean on ruby/smalltalk's `select`/`reject` to be more clear about the intended effect. 

NOTE: suggestion requires rename propagation

```suggestion
    def reject_unsupported_actions(actions)
```

Reply:
Thanks for the awesome feedback!
I usually avoid giving a naming includes `reject` (which sounds to me a sense of ending the process) but from ruby's `select` point of view, it seems `reject` is appropriative.

Reply:
In Java, lambda expressions use `filter` mostly.